problem_builder/step.py

```diff
@@ -240,6 +243,45 @@ class ReviewStepBlock(XBlockWithPreviewMixin, XBlock):
         default="Review Step"
     )
 
+    @property
+    def allowed_nested_blocks(self):
+        """
+        Returns a list of allowed nested XBlocks. Each item can be either
+        * An XBlock class
+        * A NestedXBlockSpec
+
+        If XBlock class is used it is assumed that this XBlock is enabled and allows multiple instances.
+        NestedXBlockSpec allows explicitly setting disabled/enabled state,
+        disabled reason (if any) and single/multiple instances.
+        """
+        return [
+            NestedXBlockSpec(CompletedMentoringMessageShim, boilerplate='completed'),
+            NestedXBlockSpec(IncompleteMentoringMessageShim, boilerplate='incomplete'),
+            NestedXBlockSpec(OnReviewMentoringMessageShim, boilerplate='on-review'),
+        ]
+
+    @XBlock.json_handler
+    def get_assessment_message(self, grade, suffix):
+        complete = grade['complete']
+        max_attempts_reached = grade['max_attempts_reached']
```

Comment:
@itsjeyd nit: fragile - if I do understand right `grade` is got from another ajax call, passed through a couple of JS functinos and than sent here. Modifications at each step might cause it to not contain `max_attemts_reached` or `complete`, so it will crash. Suggesting using `grade.get('max_attempts_reached')`.

Also, might make sense to add where this comes from (i.e. which AJAX handler produces the results passed here as `grade`)